### DISCUSSION TITLE
Bundles generated in development version cause "No matching version found for" errors - How to correctly do bundles on development?

### DISCUSSION THREAD

**Discussion by MTRNord** [codeoverflow-org/nodecg-io#893] [Q&A] [answered]
Since there is a node-gyp bug on linux on 0.2 of nodecg-io I have to use the development version. Which for me in itself is fine. However the bundle generated by the generate command ends with this error:

```
>>> npm install
npm ERR! code ETARGET
npm ERR! notarget No matching version found for nodecg-io-twitch-auth@^0.3.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/marcel/.npm/_logs/2023-03-11T20_08_02_293Z-debug-0.log

Couldn't generate bundle:
Command "npm install" returned error code 1!
```

what is the correct way of fixing this issue?

**Comment by MTRNord** [comment]
It seems like this exact error is fixed by just replacing it with `^0.3.0` in the versions however that breaks `nodecg-io-core` with the same error type

**Reply by MTRNord** [reply]
It seems like after some tinkering the solution is: Keep the generated stuff (dont touch it) but add this to the package.json:

```
  "overrides": {
    "nodecg-io-twitch-chat": {
      "nodecg-io-twitch-auth": "https://codeoverflow-org.github.io/nodecg-io-publish/nodecg-io-twitch-auth-0.3.0.tgz"
    },
    "nodecg-io-twitch-api": {
      "nodecg-io-twitch-auth": "https://codeoverflow-org.github.io/nodecg-io-publish/nodecg-io-twitch-auth-0.3.0.tgz"
    },
    "nodecg-io-twitch-addons": {
      "nodecg-io-twitch-auth": "https://codeoverflow-org.github.io/nodecg-io-publish/nodecg-io-twitch-auth-0.3.0.tgz"
    }
  }
```

However I wonder if there is a better solution to this

**Comment by hlxid** [comment, marked as answer]
This is a problem specific to the twitch services, other services work fine.
All twitch services depend on the utility package `nodecg-io-twitch-auth` and the package.json of each service depends on the latest development version (currently `0.3.0`). When installing the packages for these services using tarballs, npm is able to get the service package but that still points to version `0.3.0` of the util package which npm tries to get from the npm registry.
This fails because the package has not been published yet and the service package should depend on the tarball of the util package.

Your workaround with `overrides` is fine for now but we definitely should patch the `package.json` file of the twitch services to point to the current tarball instead of the unpublished version on the npm registry in our tarball build pipeline. I will make a PR to fix this hopefully this evening.
